### PR TITLE
Build: Split nightly job into its own workflow. Remove ARM64 platform builds from our triggered/on commit builds.

### PR DESCRIPTION
**Pull request by gmalouf**
## Summary
For reducing build times, improving developer workflows and some optimizations, we are moving ARM64 builds to be solely nightly jobs. As a part of this PR, we separated the nightly runs into their own dedicated workflow for easier identification/cost management.

## Test Plan
Run nightly builds - check running of triggered builds.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -203,17 +152,63 @@ workflows:
...
+          filters:
+            <<: *filters-default
+          context: slack-secrets
+          <<: *slack-fail-post-step
+
+      - tests_verification_job:
+          name: << matrix.platform >>_<< matrix.job_type >>_verification
+          matrix:
+            parameters:
+              platform: ["amd64", "arm64"]
```
Comment (onetechnical) [inline .circleci/config.yml:206]: Perhaps this instead?

```suggestion
              platform: ["amd64"]
```

**Review by onetechnical — CHANGES_REQUESTED** [review]
One change, otherwise seems pretty straightforward.

**Comment by onetechnical** [comment]
The test restrictions are just for PRs, and because you're leaving the name circleci_build_and_test I think PRs to master work as is. rel/nightly doesn't have this (no PRs for rel/nightly), so we'll just have to update the restriction on rel/beta and rel/stable once it's getting merged out to them.

**Review by onetechnical — APPROVED** [review]

File: .circleci/config.yml [hunk 2]
```diff
@@ -203,17 +152,63 @@ workflows:
...
-              only:
-                - /rel\/.*/
-                - << pipeline.parameters.valid_nightly_branch >>
           context:
             - slack-secrets
             - aws-secrets
           <<: *slack-fail-post-step
 
-      #- windows_x64_build
+  "circleci_build_and_test":
```
Comment (algobarb) [inline .circleci/config.yml:160]: we can probably remove the quotes here then too

**Review by algobarb — APPROVED** [review]
I think it looks ok to me.

**Pull request merged**